ocrd/ocrd/cli/workspace.py

```diff
@@ -21,10 +22,20 @@ def clean_id(dirty):
 
 class WorkspaceCtx():
 
-    def __init__(self, directory, mets_basename, automatic_backup):
-        self.directory = directory
+    def __init__(self, directory, mets_url, automatic_backup):
+        if directory and not mets_url:
+            mets_url = join(directory, 'mets.xml')
+        elif not directory and mets_url:
+            if mets_url.startswith('http') or mets_url.startswith('https:'):
+                raise ValueError("--mets is an http(s) URL but no --directory was given")
+            directory = dirname(mets_url) or getcwd()
+        else:
+            directory = getcwd()
+            mets_url = join(directory, 'mets.xml')
```

Comment:
I believe there is a case missing here: **both** `directory` **and** `mets_url` might be given. And as a sub-case, if `mets_url` is local and its dirname does not equal the `directory` (perhaps after canonicalization of both), then also raise an exception `"--mets has a directory part inconsistent with --directory"`.

Reply:
Indeed, thanks. a199d78